hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -698,10 +698,10 @@ private Pair<Option<String>, JavaRDD<WriteStatus>> writeToSink(JavaRDD<HoodieRec
     boolean hasErrors = totalErrorRecords > 0;
     if (!hasErrors || cfg.commitOnErrors) {
       HashMap<String, String> checkpointCommitMetadata = new HashMap<>();
-      if (checkpointStr != null) {
+      if (cfg.checkpointWriteToHoodie && checkpointStr != null) {
         checkpointCommitMetadata.put(CHECKPOINT_KEY, checkpointStr);
       }
-      if (cfg.checkpoint != null) {
+      if (cfg.checkpointWriteToHoodie && cfg.checkpoint != null) {
         checkpointCommitMetadata.put(CHECKPOINT_RESET_KEY, cfg.checkpoint);
       }
 
```

Comment:
Was the Streamer ever reading the offsets from those committed to .hoodie 
It only relied on the kafka offsets right ?
 -- every batch was reading from the available start then ?

Reply:
I guess this needs to be updated too 
https://github.com/fairmatic/hudi/blob/release-0.13.0/hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java#L597

Reply:
1. Yes, empty HashMap writes to commit files normally, it does not write the offset. It returns success too.
2. Yes, streamer always reads from the hoodie commit, except the cases where it does not find the checkpoint from the hoodie commit. We will prevent this write to hoodie commit and force it to read from kafka offset always.
3. This does not need to be changed, because it will not find the checkpoint from the hoodie commit timeline and return Option.empty(), which is fine.

Reply:
@KishanFairmatic this then assumes that no other DeltaStreamer writes /has written to the table. As that sync could have created a commit timeline..

Reply:
Thought about it. Makes sense. Making the changes.